src/lib/reasoners/shostak.ml

```diff
@@ -214,6 +238,16 @@ struct
 
   let term_embed t = hcons {v = Term t; id = -1000 (* dummy *)}
 
+  let abstract ~kind arg =
+    (* No need to nest abstractions
+       XXX: would it be OK to abstract terms into themselves? *)
+    let arg =
+      match arg.v with
+      | Abstract (_, arg) -> arg
```

Comment:
I don't think we can reach this case currently; it is more of a future-proofing thing.

Note that since we don't look inside abstracted terms, except to enforce definitional equality, it is OK to replace the inner value with any other representation provided that the two representations are definitionally equal (and `Abstract (_, arg)` is definitionally equal to `arg`).